packages/@aws-cdk/cdk/lib/stack.ts

```diff
@@ -179,41 +181,21 @@ export class Stack extends Construct implements ITaggable {
   }
 
   /**
-   * Returns the CloudFormation template for this stack by traversing
-   * the tree and invoking _toCloudFormation() on all Entity objects.
-   *
-   * @internal
+   * Resolve a tokenized value in the context of the current Construct
    */
-  public _toCloudFormation() {
-    // before we begin synthesis, we shall lock this stack, so children cannot be added
-    this.node.lock();
-
-    try {
-      const template: any = {
-        Description: this.templateOptions.description,
-        Transform: this.templateOptions.transform,
-        AWSTemplateFormatVersion: this.templateOptions.templateFormatVersion,
-        Metadata: this.templateOptions.metadata
-      };
-
-      const elements = cfnElements(this);
-      const fragments = elements.map(e => this.node.resolve(e._toCloudFormation()));
-
-      // merge in all CloudFormation fragments collected from the tree
-      for (const fragment of fragments) {
-        merge(template, fragment);
-      }
-
-      // resolve all tokens and remove all empties
-      const ret = this.node.resolve(template) || {};
-
-      this.logicalIds.assertAllRenamesApplied();
+  public resolve(obj: any): any {
+    return resolve(obj, {
+      scope: this,
+      prefix: [],
+      resolver: CLOUDFORMATION_TOKEN_RESOLVER,
+    });
+  }
 
-      return ret;
-    } finally {
-      // allow mutations after synthesis is finished.
-      this.node.unlock();
-    }
+  /**
+   * Convert an object, potentially containing tokens, to a JSON string
+   */
+  public stringifyJson(obj: any): string {
```

Comment:
This does not *stringify JSON*, it *stringifies stuff to CFN-y Json*. I don't love that method name, and I kinda think `stringify` or `toJsonString` is better.

Reply:
You are right. I like `toJsonString`.

Reply:
@rix0rrr what do you think?

Reply:
Yeah cool